maestro-client/src/main/java/maestro/drivers/IOSDriver.kt

```diff
@@ -373,7 +378,48 @@ class IOSDriver(
     override fun backPress() {}
 
     override fun hideKeyboard() {
-        pressKey(KeyCode.ENTER)
+        swipe(
+            start = Point(widthPercentToPoint(0.5), heightPercentToPoint(0.5)),
+            end = Point(widthPercentToPoint(0.5), heightPercentToPoint(0.47)),
+            durationMs = 50,
+        )
+
+        if (isKeyboardHidden()) {
+            return
+        }
+
+        swipe(
+            start = Point(widthPercentToPoint(0.5), heightPercentToPoint(0.5)),
+            end = Point(widthPercentToPoint(0.47), heightPercentToPoint(0.5)),
+            durationMs = 50,
+        )
+
+        if (isKeyboardHidden()) {
+            return
+        }
```

Comment:
I imagined this to be implemented in the XCTest runner.
The runner can create a single EventRecord with both the swipe down and up in there. Reducing the change that the `hidekeyboard` command changes anything in the app: Separate swipe events are more likely to trigger some app logic than a single swipe down then up event.

Reply:
Although, on second thought.
For buttons to not be tapped, the touch point needs to be swiped outside of the tappable area. If the user swipes back to the tappable area, the press will not be cancelled.
Did you test with swipe down event right on a button?
A quick check with the cancel button in Setting Search app, the swipe needs to be roughly 1/5th the screen height before it is considered cancelled.

Reply:
This is the change I did:

```
func addHideKeyboardSwipeEvent(start: CGPoint, end: CGPoint, duration: TimeInterval) -> Self {
        var path = PointerEventPath.pathForTouch(at: start)
        path.offset += Self.defaultTapDuration
        path.moveTo(point: end)
        path.moveTo(point: start)
        path.moveTo(point: end)
        path.offset += duration
        path.liftUp()
        return add(path)
}
```


## EventRecord Changes

https://user-images.githubusercontent.com/12881364/229527384-cd15d780-e5f0-4f40-9828-f8eb93a270f1.mov

## Current Implementation

https://user-images.githubusercontent.com/12881364/229528802-47718aac-e06a-4ca6-ad59-c8b9bb847f98.mov


Seems like same, unless I am missing something in that eventRecord code.